### PR TITLE
rejig

### PR DESCRIPTION
**Pull request by dan-jones-nimble**
rejig login page to libs, add loggedInContext to be shared across apps and web

### CODE REVIEW

File: libs/location/src/index.ts [hunk 1]
```diff
@@ -34,6 +39,6 @@ defineTask(LOCATION_TASK_NAME, ({ data, error }) => {
   if (data) {
     const { locations } = data;
     console.log(locations);
```
Comment (ghost) [inline libs/location/src/index.ts:56]: Keeping this ?

File: libs/components/native/Screens/Login/index.tsx [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+      [name]: value
+    }));
+  };
+
+  const submitCredentials = async () => {
+    const attemptedEmailLogin = await emailLogin({
+      email: credentials.email,
+      password: credentials.password
+    });
+    if (attemptedEmailLogin?.$id) logIn();
```
Comment (ghost) [inline libs/components/native/Screens/Login/index.tsx]: Want any sort of error messaging?

File: libs/location/src/index.ts (hunk 1 above)
Comment (dan-jones-nimble) [reply libs/location/src/index.ts:56]: think so for now

File: libs/components/native/Screens/Login/index.tsx (hunk 2 above)
Comment (dan-jones-nimble) [reply libs/components/native/Screens/Login/index.tsx]: working on it :+1:

File: libs/components/native/Screens/Login/index.tsx [hunk 3]
```diff
@@ -0,0 +1,80 @@
+// Import
```
Comment (ghost) [inline libs/components/native/Screens/Login/index.tsx:1]: Does the wobble still trigger is a user clicks the button twice in a row without a change in the user input?

File: libs/location/src/index.ts [hunk 4]
```diff
@@ -1,39 +1,58 @@
...
+};
+
+export const startLocationTracking = async () => {
+  if (Platform.OS !== 'web') {
+    await startLocationUpdatesAsync(BACKGROUND_LOCATION_TRACKING_TASK_NAME, {
+      accuracy: LocationAccuracy.High,
+      distanceInterval: 5
+    });
+  } else {
+    throw new Error('Location tracking not available on Web.');
```
Comment (ghost) [inline libs/location/src/index.ts:39]: Throwing an error on web, does it disrupt the user with an error message?

File: libs/components/native/Screens/Login/index.tsx (hunk 3 above)
Comment (dan-jones-nimble) [reply libs/components/native/Screens/Login/index.tsx:1]: no the user has to change input :/

Comment (dan-jones-nimble) [reply libs/components/native/Screens/Login/index.tsx:1]: wasnt motivated enough to sort out - can have another look if u want

File: libs/location/src/index.ts (hunk 4 above)
Comment (dan-jones-nimble) [reply libs/location/src/index.ts:39]: the error is caught a level above

File: apps/door-to-door-app/utils/index.ts [hunk 5]
```diff
@@ -0,0 +1,7 @@
+import { randomUUID } from 'expo-crypto';
+import { storeRouteId } from '@nx-expo/storage';
+
+export const newRouteId = () => {
```
Comment (ghost) [inline apps/door-to-door-app/utils/index.ts:4]: Appwrite provides a UID generator, no?

**Review by ghost — APPROVED** [review]

**Pull request merged**